<!doctype html>
<!-- !doctype html public "-//W3C//DTD HTML 4.0 Transitional //EN"> -->
<html>
  <head>
    <meta http-equiv="content-type" content="text/html; charset=utf8" /> 
    <title>An Empirical Justification Of False Discovery Rate Correction With Small Samples</title>
    
    <meta name="author"  content="Stuart Andrews"/>

    <!-- 2011-03-21 START CUSTOM HEADER -->

    <!-- Stylesheet: Pygments -->
    <link href='/css/pygments.css' type='text/css' rel='stylesheet'/>

    <!-- Stylesheet: Homepage -->
    <link href='/css/style-combined.css' type='text/css' rel='stylesheet'/>

    <!-- MATHJAX hosted CDN.MATHJAX.ORG -->
    <script type='text/javascript' src='http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML'> MathJax.Hub.Config({ extensions: ["tex2jax.js","MathMenu.js","MathZoom.js"], jax: ["input/TeX","output/HTML-CSS"], tex2jax: {inlineMath: [["$","$"], ["\\(","\\)"]], skipTags: ["script","noscript","style","textarea","pre"], processEnvironments: true}, TeX: { extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"] } }); </script>

    <!-- GOOGLEAPIS hosted JQUERY.MIN -->
    <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js"></script>

    <!-- GITHUB hosted JQUERY.COOKIE.JS -->
    <script type="text/javascript" src="https://raw.github.com/carhartl/jquery-cookie/master/jquery.cookie.js"></script>

    <!-- DAYTIME/NIGHTTIME JAVASCRIPT -->
    <script type="text/javascript" src="/css/daynight.js"></script>
    <!-- 2011-03-21 END CUSTOM HEADER -->
  </head>
  <body>
    <!-- START BODY -->
    <div class="site">
      <div class="header">
        <span class="title"><a href="/">stuartjandrews.com</a></span>
        <button id="daynight">Day/Night</button>
        <!-- <hr> -->
      </div>

      <div class="content">
        <h1>An Empirical Justification Of False Discovery Rate Correction With Small Samples</h1>

<div id="post">
<!--
[sim.t.test.density-p-value-p1.jpg]: http://lh3.ggpht.com/-AHYz5NXycB0/TsRmvcFx40I/AAAAAAAABLE/i-OFk4ttoMc/sim.t.test.density-p-value-p1.jpg =340x "Figure 1"
[sim.t.test.density-p-value-p2.jpg]: http://lh5.ggpht.com/-wJop5cKhbjo/TsRmwHgEGII/AAAAAAAABLI/dnim9avBhAE/sim.t.test.density-p-value-p2.jpg =340x "Figure 2"
[sim.t.test.density-p-value-p3.jpg]: http://lh3.ggpht.com/-uSfKnS_-WRQ/TsRmwxz7EBI/AAAAAAAABLM/yKqLXcC4RwI/sim.t.test.density-p-value-p3.jpg =340x "Figure 3"
[sim.t.test.density-p-value-p4.jpg]: http://lh4.ggpht.com/-OY15_MfRzdE/TsRmxXjKp0I/AAAAAAAABLQ/_EF_TaGBlyw/sim.t.test.density-p-value-p4.jpg =340x "Figure 4"
# Low-Hanging Fruit
> %justcenter%
> ![sim.t.test.density-p-value-p1.jpg][]
>
> Figure 1
>
> ![sim.t.test.density-p-value-p2.jpg][]
>
> Figure 2
<div class="justcenter"> </div>
-->

<p>A central problem in genomics is the identification of the molecular machinery that is responsible for the phenotype, the set of all observable characteristics, of an organism.  One of the more informative sources of evidence is <em>gene expression</em>, which is the abundance of molecules that a gene produces in a given context.  The advent of high-throughput technologies and related statistical methods provide a means to pick the most probable candidate genes - the &quot;low-hanging fruit&quot; - that determine phenotype.</p>

<ol>
<li>High-throughput technologies enable measurement of the <em>differential expression</em> (DE) between two groups or conditions for a large collection of genes, and </li>
<li>Statistical methods, known as false discovery rate (FDR) correction, help to determine which of the measured differences are truly significant.  These methods play an important role in cost-benefit analysis when deciding how many potential DE genes to select for validation.</li>
</ol>

<!-- This post examines the behaviour of **p-values** and the **false discovery rate (FDR)** for small samples. -->

<p>In the following, a simulation study involving randomly generated <em>gene expressions</em> for a large number of genes is used to analyze the behaviour of <em>p-values</em> and the influence of <em>false discovery rate (FDR) correction</em> with small sample sizes.</p>

<h1 id="toc_44">Hypothesis Testing</h1>

<p>Hypothesis testing is used to decide whether an observed DE gene occured by chance (&quot;the null&quot;), or constitutes evidence of an underlying biological effect (&quot;the alternative&quot;).  The p-value for a hypothesis test is the chance of observing measurements as extreme as found in the experiment under the assumption that there is no difference in the two conditions.  It is closely related, but not necessarily equal, to the probability of false discovery, which is the probability that the gene is falsely classified as DE.</p>

<p>Suppose there are $ N $ samples each from conditions $ A $ and $ B $.  The magnitude of DE is the difference between the two means, $ \bar{x}_A-\bar{x}_B $.  If the samples are independent, normally-distributed, and the variances of the two groups are the same, then a suitable test is the T-test.  A t-statistic is calculated as</p>

<p>$$
  T =  \frac{\bar{x}_B - \bar{x}_A}{s\sqrt{2/N}}
$$</p>

<p>where $ s $ is the pooled estimate of the standard deviation.  Under the assumptions of the test, the value $ T\, $ follows a Student&#39;s t-distribution, with $ d = 2 * N - 2 $ degrees of freedom.  The p-value for the 2-sided T-test is computed as</p>

<p>$$
  p = 1 - P\left( -|T| &lt; t &lt; |T| \right) .
$$</p>

<p>If the p-value is less than the level-of-significance $ \alpha $, then the null hypothesis is rejected, and the observed DE is significant at that $ \alpha $-significance level.</p>

<h2 id="toc_45">Example</h2>

<p>Suppose there are $ N=4 $ samples, each, from both conditions with the following gene expression values:  $ A = \left( 2, 4, 1, 1 \right) $ and $ B = \left( 9, 11, 10, 7 \right) $.  The t-statistic is calculated as [ttest-eg] where $ s \approx 4.14\,$ is the pooled estimate of the standard deviation.  Using the lookup table on <a href="http://en.wikipedia.org/wiki/Student%27s_t-distribution">Student&#39;s t-distribution</a>, the p-value for this 2-sided test is $ p &lt; 0.05 $ supporting the conclusion that significant DE was observed.</p>

<p>$$
  T = \frac{(9 + 11 + 10 + 7)/4 - (2 + 4 + 1 + 1)/4}{s \sqrt{2 / 4}} \approx 2.478\,,
$$</p>

<!-- x = np.array([[2,4,1,1],[9,11,10,7]]); np.diff(np.mean(x, axis=1))*sqrt(2)/(np.std(x, ddof=1)) -->

<h1 id="toc_46">False Discovery Rate Correction</h1>

<p>When testing a large number of genes, a threshold based on the p-value fails to meaningfully convey the significance of the resulting list of genes because it does not take into account the elevated chance of observing extreme levels of differential expression through repeated, independent trials.  So, for example, while it is unusual to observe a p-value of 0.01 in a single experiment, it is nevertheless expected that such small p-values will be observed with regularity as more and more genes are tested. </p>

<p>In order to provide a meaningful measure of significance, methods have been developed to estimate the false discovery rate for the top-k most DE genes.  For example, the Benjamini-Hochbergâ€“Yekutieli (BH) method estimates the FDR to be the minimum $ q_k $ such that [qval-eqn] where $ p_i $ is the $ i $-th smallest p-value, and $ m $ is the number of genes tested.  The value $ q_k $ for $ k = 1, \ldots, m $, associated with the gene at rank k, is referred to as the FDR-corrected p-value.</p>

<p>$$
  p_i \leq \frac{i}{m} q_k, \quad \forall i\leq k \,. 
$$</p>

<h1 id="toc_47">Genomics Simulation</h1>

<p>A simulation study involving randomly generated gene expressions for a large number of genes is used to analyze the behaviour of <em>p-values</em> and the influence of <em>false discovery rate (FDR) correction</em> with small sample sizes.  This setting was chosen to resemble the genomics context.  Specifically, a large number of genes are simultaneously screened for differential expression between groups, using only a small number of samples.</p>

<ol>
<li>Each data set, numbered by $ i = 1, \ldots, 2000 $, is comprised of $ N $ randomly generated samples from each of two groups $ A_i $ and $ B_i $.</li>
<li>Samples are assumed to be normally-distributed: $ A_i $&#39;s samples have mean $ \mu_i $, $ B_i $&#39;s samples have mean $ \nu_i $, and both have variance $\mathrm{sqr}(\sigma_i) = 1$.</li>
<li>With fixed values of the free parameters: $ N $ , $ \mu_i $ , and $ \nu_i $, a p-value is calculated as above for each comparison, giving rise to a distribution across all 2000 tests.</li>
</ol>

<p>In the figures below, the shape and variability of the distribution of p-values is analyzed.  Each figure contains three curves corresponding to data sets of varying size: red($ N=3 $), green($ N=5 $), and blue($ N=7 $).  Due to the procedure used for density estimation, the displayed curves extend beyond the unit interval, despite the fact that p-values are bounded $ 0 \leq p_i \leq 1 $.  While in common practice there will be a mixture of: 1) DE, and 2) non-DE genes in a single experiment, these two cases are treated separately for clarity.</p>

<h2 id="toc_48">DE &amp; Non-DE Genes Without FDR Correction</h2>

<p>Figure 1 shows the distribution of p-values when the groups are differentially expressed, with $ \mu_i = 0 $ and $ \nu_i = 1 $ for all $ i $ .  By design, a large proportion of p-values are close to zero, and for $ N \geq 7 $,  below the commonly-used $ \alpha = 0.05 $ level-of-significance.  However, each distribution has a long tail that extends toward $ p_i = 1 $.  This effect is most pronounced for $ N=3 $, which is a reflection of the diminishing power of the T-test as $ N $ decreases.  Thus, for $ N \geq 7 $, it is relatively easy to detect differential expression.</p>

<p>Figure 1 &amp; 2: <img src="sim.t.test.density-p-value-p1.jpg" alt="Figure_1" title="Figure 1"> <img src="sim.t.test.density-p-value-p2.jpg" alt="Figure_2" title="Figure 2"></p>

<p>Figure 2 shows the distribution of p-values when groups $ A_i $ and $ B_i $ have the same mean $ \mu_i = \nu_i $ for all $ i $, and so are not differentially expressed.  The p-values are approximately uniformly distributed, in accordance with a classical result in statistics.  Roughly speaking, since under the null-hypothesis (non-DE) the test statistic follows a Student&#39;s-t distribution, the p-value is obtained from its cumulative distribution function, and is therefore uniform.  See <a href="http://stats.stackexchange.com/questions/10613/why-p-values-are-uniformly-distributed">here</a>, <a href="http://en.wikipedia.org/wiki/Probability_integral_transform">here</a>, and <a href="http://pubs.amstat.org/doi/abs/10.1198/000313008X332421">here</a> for details.</p>

<p>The take-home message of this simulation is that one observes $ p_i \leq 0.05 $ about 5% of the time for genes that are not differentially expressed.  This result demonstrates why statistical methods have been developed to compensate for false positives in multiple testing scenarios.</p>

<h2 id="toc_49">DE &amp; Non-DE Genes With FDR Correction</h2>

<p>Figures 3 and 4 show the effect of FDR-correction on the distributions of p-values from Figures 1 and 2, respectively.  Again, each figure contains three curves corresponding to data sets of varying size: red($ N=3 $), green($ N=5 $), and blue($ N=7 $).  The dominant effect is that each distribution has been shifted toward $ p=1 $, lowering the estimated likelihood of discovery for each associated gene. </p>

<p>For example, for $ N=3 $ in Figure 3, the red curve shows an estimated false discovery rate of between 10-20% amongst the first selected DE genes.  Fortunately, as shown in Figure 4, non-DE genes are assigned FDR-corrected p-values close to 1 thereby diminishing the chance that these genes will be selected. </p>

<p>Figure 3 &amp; 4: <img src="sim.t.test.density-p-value-p3.jpg" alt="Figure_3" title="Figure 3"> <img src="sim.t.test.density-p-value-p4.jpg" alt="Figure_4" title="Figure 4"></p>

<p>Caveat 
: In practice, FDR-correction is applied to the mixture of p-values from DE and non-DE genes.  Figures 3 and 4 do not accurately represent the behaviour of FDR-correction on this mixture, because each corrected p-value depends on the input collection to which it belongs.  A more informative result could be obtained by performing FDR-correction on the entire set of simulated genes, DE and non-DE, before segregating and plotting.</p>

<h1 id="toc_50">Summary</h1>

<p>This post analyzed the behaviour of p-values and the influence of FDR-correction in a simulation study with many genes but only a few samples from each group.  Simulation studies such as this one are useful for illuminating the cost-benefit trade-offs encountered in the search for molecular determinants of phenotype with high-throughput screening.</p>

<h1 id="toc_51">Learn More</h1>

<ul>
<li><a href="http://en.wikipedia.org/wiki/Genomics">Genomics</a></li>
<li><a href="http://en.wikipedia.org/wiki/Gene_Expression">Gene Expression</a></li>
<li><a href="http://en.wikipedia.org/wiki/Student%27s_t-test">Student&#39;s t-test</a></li>
<li><a href="http://en.wikipedia.org/wiki/Student%27s_t-distribution">Student&#39;s t-distribution</a></li>
<li><a href="http://en.wikipedia.org/wiki/P-value">P-value</a></li>
<li><a href="http://en.wikipedia.org/wiki/Multiple_comparisons">Multiple comparisons</a></li>
<li><a href="http://en.wikipedia.org/wiki/False_Discovery_Rate">False Discovery Rate</a></li>
</ul>

</div>

<div id="related">
  <h2>Related Posts</h2>
  <ul class="posts">
    
    <li><span>24 Sep 2012</span> &raquo; <a href="/2012/09/24/NGS-ML-Intro.html">An Introduction to Machine Learning for Biologists</a></li>
    
    <li><span>10 Jun 2011</span> &raquo; <a href="/2011/06/10/Wiki-Graph.html">Wiki Graph - A Visualization Tool For Wikis</a></li>
    
    <li><span>06 Sep 2011</span> &raquo; <a href="/2011/09/06/Hadoop-Filesystem-Tools.html">Hadoop Filesystem Tools</a></li>
    
    <li><span>17 May 2011</span> &raquo; <a href="/2011/05/17/A-Centralized-VCS-For-Config-Files.html">A Centralized VCS For Config Files</a></li>
    
  </ul>
</div>

<hr>




  <div id="disqus_thread"></div>
<script type="text/javascript">
    var disqus_developer = 1;
    var disqus_shortname = 'istructure'; // required: replace example with your forum shortname
    
    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = 'http://' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com" class="dsq-brlink">blog comments powered by <span class="logo-disqus">Disqus</span></a>





<div class="span4">
  
    <h4>Tags</h4>
    <ul class="tag_box">
    
    


  
     
    	<li><a href="/tags.html#Gene-ref">Gene <span>1</span></a></li>
     
    	<li><a href="/tags.html#Genomics-ref">Genomics <span>1</span></a></li>
     
    	<li><a href="/tags.html#Biology-ref">Biology <span>1</span></a></li>
     
    	<li><a href="/tags.html#Rstats-ref">Rstats <span>1</span></a></li>
     
    	<li><a href="/tags.html#Ttest-ref">Ttest <span>1</span></a></li>
    
  



    </ul>
  
  <h4>Published</h4>
  <div class="date"><span>16 November 2011</span></div>
</div>



      </div>
      <div class="footer">
        <div class="contact">
          <hr>
          <p>
          by Stuart Andrews
          <a href="http://github.com/tub78/">@tub78 (github)</a>
          <a href="http://twitter.com/tub78/">@tub78 (twitter)</a>
          <span class="tiny">[<a href="/about.html">About this blog</a>]</span>
          </p>
        </div>
      </div>
    </div> <!-- site -->

    
    <!-- END BODY -->
  </body>
</html>

